cmd/dockerd/daemon.go

```diff
@@ -244,6 +254,30 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 		logrus.Fatalf("Error starting cluster component: %v", err)
 	}
 
+	builderStateDir := filepath.Join(cli.Config.Root, "builder")
+
+	fsCache, err := fscache.NewFSCache(fscache.Opt{
+		Backend: fscache.NewNaiveCacheBackend(builderStateDir),
+		Root:    builderStateDir,
+		GCPolicy: fscache.GCPolicy{ // TODO: expose this in config
```

Comment:
I tried to do that but noticed that the current logic does not allow adding configuration options without adding a cli flag as well. So as this PR is already quite large I'd prefer to not include that refactoring here as well.